public/app/features/dashboard/components/GenAI/GenAIDashDescriptionButton.tsx

```diff
@@ -45,7 +45,7 @@ function getMessages(): Message[] {
     },
     {
       content: `The title of the dashboard is "${dashboard.title}"\n` + `${panelPrompt}`,
-      role: Role.system,
+      role: Role.user,
```

Comment:
I am wondering if we should modify this as a user prompt vs appending it to the above system prompt? I believe changing classification of role on this may modify current behavior / functionality which we may want to avoid with this reformatting PR 🤷

Reply:
So there's 2 reasons: 
1. this part of the prompt does look more like a user prompt (e.g. context-specific input) rather than a system prompt (e.g. instructions) + fits with our new guidance (at least 1 user message + 1 optional system message) to make sure we standardize our approach to work with multiple LLM models.

2. The difference in performance is between the versions (`main` vs this PR) are minimal for OpenAI models, to be sure I ran the benchmark on a larger sample:

## Prompt: prompts/panel-title/v0, N=841, temp=1.0
| model                  |   main |   PR |
|:-----------------------|--------------------------:|------------------------:|
| gpt-4o-2024-05-13      |                     0.816 |                   0.813 |
| gpt-4-turbo-2024-04-09 |                     0.798 |                   0.794 |
| gpt-3.5-turbo-0125     |                     0.777 |                   0.778 |


## Prompt: prompts/dashboard-description/v0, N=506, temp=1.0
| model                  |   main |   PR |
|:-----------------------|--------------------------:|------------------------:|
| gpt-4-turbo-2024-04-09 |                     0.697 |                   0.698 |
| gpt-4o-2024-05-13      |                     0.694 |                   0.696 |
| gpt-3.5-turbo-0125     |                     0.608 |                   0.613 |



## Prompt: prompts/panel-description/v1, N=841, temp=1.0
| model                  |   main |   PR |
|:-----------------------|--------------------------:|------------------------:|
| gpt-4-turbo-2024-04-09 |                     0.782 |                   0.778 |
| gpt-4o-2024-05-13      |                     0.781 |                   0.776 |
| gpt-3.5-turbo-0125     |                     0.771 |                   0.763 |

## Prompt: prompts/dashboard-title/v0, N=506, temp=1.0
| model                  |   main |   PR |
|:-----------------------|--------------------------:|------------------------:|
| gpt-4-turbo-2024-04-09 |                     0.671 |                   0.669 |
| gpt-4o-2024-05-13      |                     0.662 |                   0.661 |
| gpt-3.5-turbo-0125     |                     0.622 |                   0.636 |

Reply:
> 1. this part of the prompt does look more like a user prompt (e.g. context-specific input) rather than a system prompt (e.g. instructions) + fits with our new guidance (at least 1 user message + 1 optional system message) to make sure we standardize our approach to work with multiple LLM models.

I think in these cases that the prompt is more of a system prompt as the `panelPrompt` is basically a continuation of instructions (see [here](https://github.com/grafana/grafana/blob/6d98f53e8b12f1cfa0166e0c46db1059d13bf8e6/public/app/features/dashboard/components/GenAI/utils.ts#L137)) 🤔 

re: 1 optional system prompt + 1 user prompt is this a hard and fast rule or more of adherence to best practices? 

Given the semantic performance isn't really impacted I'm ok with deferring to your judgement on this :)